paymentsheet/src/main/java/com/stripe/android/paymentsheet/ui/AddPaymentMethodInteractor.kt

```diff
@@ -76,13 +76,14 @@ internal class DefaultAddPaymentMethodInteractor(
     companion object {
         fun create(sheetViewModel: BaseSheetViewModel): AddPaymentMethodInteractor {
             val formHelper = FormHelper.create(sheetViewModel)
+            val paymentMethodMetadata = requireNotNull(sheetViewModel.paymentMethodMetadata.value)
```

Comment:
I think in the old behavior, if paymentMethodMetadata wasn't loaded yet, then we would return empty list for the supported PMs, but not we would crash. Can we maintain that behavior?

And then ooc -- long term, will we be updating this function to just take in a non-null paymentMethodMetadata and only call this function once paymentMethodMetadata has been loaded?

Reply:
You can't get here if PaymentMethodMetadata is null. And we wouldn't want to make this an empty list, since that would be a bug, and the user would be blocked from checking out anyway.

We also do this in vertical mode: https://github.com/stripe/stripe-android/blob/a9afb9810c76c19d49a72ab084bf5cac31482bdf/paymentsheet/src/main/java/com/stripe/android/paymentsheet/verticalmode/PaymentMethodVerticalLayoutInteractor.kt#L89

I think this is the right move for right now, but like you said, long term we will pass in a non null `PaymentMethodMetadata` here, rather than relying on a nullable flow.